test/dates/types.jl

```diff
@@ -168,6 +192,12 @@ ms = Dates.Millisecond(1)
 @test Dates.Date(d,y) == Dates.Date(1,1,1)
 @test Dates.Date(d,m) == Dates.Date(1,1,1)
 @test Dates.Date(m,y) == Dates.Date(1,1,1)
-
-@test isfinite(Dates.Date)
-@test isfinite(Dates.DateTime)
+us = Dates.Microsecond(1)
+ns = Dates.Nanosecond(1)
+@test Dates.Time(h) == Dates.Time(1)
+@test Dates.Time(h,mi) == Dates.Time(1,1)
+@test Dates.Time(h,mi,s) == Dates.Time(1,1,1)
+@test Dates.Time(h,mi,s,ms) == Dates.Time(1,1,1,1)
+@test Dates.Time(h,mi,s,ms,us) == Dates.Time(1,1,1,1,1)
+@test Dates.Time(h,mi,s,ms,us,ns) == Dates.Time(1,1,1,1,1,1)
+@test Dates.Time(us,h,s,ns,mi,ms) == Dates.Time(1,1,1,1,1,1)
```

Comment:
You're missing the point of what's lacking about this. The first two places you linked to have the same basic issue of using zeros for all but one input. The third doesn't test this direction of taking time typed objects as inputs to this constructor.